src/workerd/api/kv.c++

```diff
@@ -310,6 +310,62 @@ jsg::Promise<jsg::Value> KvNamespace::list(jsg::Lock& js, jsg::Optional<ListOpti
   });
 }
 
+namespace {
+kj::Promise<void> handlePut(auto& context,
+                            auto client,
+                            auto urlStr,
+                            auto headers,
+                            auto expectedBodySize,
+                            auto supportedBody) {
+  co_await context.waitForOutputLocks();
+  auto innerReq = client->request(kj::HttpMethod::PUT, urlStr, headers, expectedBodySize);
+
+  struct RefcountedWrapper: public kj::Refcounted {
+    explicit RefcountedWrapper(kj::Own<kj::HttpClient> client): client(kj::mv(client)) {}
+    kj::Own<kj::HttpClient> client;
+  };
+  auto rcClient = kj::refcounted<RefcountedWrapper>(kj::mv(client));
+  // TODO(perf): More efficient to explicitly attach rcClient below?
+  // TODO(cleanup): Is attaching here necessary in the coroutine?
+  auto req = attachToRequest(kj::mv(innerReq), kj::mv(rcClient));
+
+  kj::Promise<void> writePromise = nullptr;
```

Comment:
maybe drop writePromise and just put co_await at write() call? Will make it easier to change too.